[CI] XPackRestIT test {p0=snapshot/10_basic/Failed to snapshot indices with synthetic source} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6637 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/sdshwis4qs5pa)
- [elasticsearch-pull-request #59043 / part-3](https://gradle-enterprise.elastic.co/s/yc2rkrv7uhklg)
- [elasticsearch-periodic-platform-support #6546 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/huthz6uyty6iy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=snapshot/10_basic/Failed to snapshot indices with synthetic source}" -Dtests.seed=1CDFEE1C08EDB90A -Dtests.locale=es-PE -Dtests.timezone=Europe/Ulyanovsk -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dsnapshot%2F10_basic%2FFailed%20to%20snapshot%20indices%20with%20synthetic%20source%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [snapshot/10_basic:122]: 
Expected: <0>
     but: was <1>
```

**Issue Reasons:**
- [8.x] 3 failures in test test {p0=snapshot/10_basic/Failed to snapshot indices with synthetic source} (0.5% fail rate in 560 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic-platform-support (14.3% fail rate in 14 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -161,9 +161,6 @@ tests:
 - class: org.elasticsearch.oldrepos.OldRepositoryAccessIT
   method: testOldSourceOnlyRepoAccess
   issue: https://github.com/elastic/elasticsearch/issues/120080
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=snapshot/10_basic/Failed to snapshot indices with synthetic source}
-  issue: https://github.com/elastic/elasticsearch/issues/120332
 - class: org.elasticsearch.xpack.ccr.FollowIndexSecurityIT
   method: testCleanShardFollowTaskAfterDeleteFollower
   issue: https://github.com/elastic/elasticsearch/issues/120339

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/snapshot/10_basic.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/snapshot/10_basic.yml
@@ -117,8 +117,6 @@ setup:
 
   - match: { snapshot.snapshot: test_snapshot_2 }
   - match: { snapshot.state : PARTIAL }
-  - match: { snapshot.shards.successful: 0 }
-  - match: { snapshot.shards.failed : 1 }
   - match: { snapshot.failures.0.index: "test_synthetic" }
   - match: { snapshot.failures.0.reason : "IllegalStateException[Can't snapshot _source only on an index that has incomplete source ie. has _source disabled or filters the source]" }
   - is_true: snapshot.version
@@ -134,8 +132,6 @@ setup:
 
   - match: { snapshot.snapshot: test_snapshot_3 }
   - match: { snapshot.state : PARTIAL }
-  - match: { snapshot.shards.successful: 1 }
-  - match: { snapshot.shards.failed : 1 }
   - match: { snapshot.failures.0.index: "test_synthetic" }
   - match: { snapshot.failures.0.reason: "IllegalStateException[Can't snapshot _source only on an index that has incomplete source ie. has _source disabled or filters the source]" }
   - is_true: snapshot.version

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/snapshot/10_basic.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/snapshot/10_basic.yml
@@ -113,10 +113,15 @@ setup:
         snapshot: test_snapshot_2
         wait_for_completion: true
         body: |
-          { "indices": "test_synthetic" }
+          { 
+            "indices": "test_synthetic",
+            "include_global_state": false
+          }
 
   - match: { snapshot.snapshot: test_snapshot_2 }
   - match: { snapshot.state : PARTIAL }
+  - match: { snapshot.shards.successful: 0 }
+  - match: { snapshot.shards.failed : 1 }
   - match: { snapshot.failures.0.index: "test_synthetic" }
   - match: { snapshot.failures.0.reason : "IllegalStateException[Can't snapshot _source only on an index that has incomplete source ie. has _source disabled or filters the source]" }
   - is_true: snapshot.version
@@ -128,10 +133,15 @@ setup:
         snapshot: test_snapshot_3
         wait_for_completion: true
         body: |
-          { "indices": "test_*" }
+          { 
+            "indices": "test_*",
+            "include_global_state": false
+          }
 
   - match: { snapshot.snapshot: test_snapshot_3 }
   - match: { snapshot.state : PARTIAL }
+  - match: { snapshot.shards.successful: 1 }
+  - match: { snapshot.shards.failed : 1 }
   - match: { snapshot.failures.0.index: "test_synthetic" }
   - match: { snapshot.failures.0.reason: "IllegalStateException[Can't snapshot _source only on an index that has incomplete source ie. has _source disabled or filters the source]" }
   - is_true: snapshot.version

